quixstreams/topic_manager.py

```diff
@@ -0,0 +1,770 @@
+import logging
+import warnings
+import pprint
+from abc import abstractmethod
+from typing import Dict, List, Mapping, Optional, Set, Literal, Protocol, ClassVar
+
+from quixstreams.platforms.quix import QuixKafkaConfigsBuilder
+from .kafka.admin import Admin
+from .models.serializers import DeserializerType, SerializerType
+from .models.topics import Topic, TopicConfig, TopicList, TopicMap
+
+logger = logging.getLogger(__name__)
+
+__all__ = ("TopicManager",)
+
+
+def dict_values(d: object) -> List:
+    """
+    Recursively unpacks a set of nested dicts to get a flattened list of leaves,
+    where "leaves" are the first non-dict item.
+
+    i.e {"a": {"b": {"c": 1}, "d": 2}, "e": 3} becomes [1, 2, 3]
+
+    :param d: initially, a dict (with potentially nested dicts)
+
+    :return: a list with all the leaves of the various contained dicts
+    """
+    if d:
+        if isinstance(d, dict):
+            return [i for v in d.values() for i in dict_values(v)]
+        elif isinstance(d, list):
+            return d
+        return [d]
+    return []
+
+
+def affirm_ready_for_create(topics: TopicList):
```

Comment:
I agree overall; I used this language on purpose because I basically didn't want to mix it up with `validating` the topics (aka checking they exist), which is a different step altogether. Maybe it's not really worth the trouble distinguishing it to that level?